forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java

```diff
@@ -0,0 +1,174 @@
+package unipd.se18.ocrcamera.forum;
+
+import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.ViewModelProvider;
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import unipd.se18.ocrcamera.forum.viewmodels.Login_VM;
+
+
+/**
+ * A fragment where a user can login to the forum service
+ *
+ * @author Leonardo Rossi (g2), Alberto Valente (g2), Taulant Bullaku (g2)
+ */
+public class ForumLogin extends Fragment {
+
+    /**
+     * ***************************
+     * **   GLOBAL VARIABLES    **
+     * ***************************
+     */
+
+    private OnFragmentInteractionListener mListener;
+
+    /**
+     * String used for logs to identify the fragment throwing it
+     */
+    private final String LOG_TAG = "@@ForumLogin";
+
+    /**
+     * String used to indicate that no username has been passed
+     */
+    private final String LOG_NULL_USERNAME = "A NULL username passed through liveLoginResponse";
+
+    /**
+     * String used to indicate that no error message has been passed
+     */
+    private final String LOG_NULL_ERROR_MESSAGE = "A NULL error message passed through liveError";
+
+    /**
+     * Key to identify the username passed to ShowPosts instance
+     */
+    private final String KEY_USERNAME = "username";
+
+    private EditText usernameEditText;
+    private EditText pwdEditText;
+    private Button loginButton;
+
+    private String userName;
+    private String userPwd;
+
+    private Login_VM viewModel;
```

Comment:
The global variables are very well commented and easy to understand, except this one (viewModel). I had to look for its class in the other files to know what it was for.

Reply:
thanks for the clarification, I will comment in more detail. thank you